### PR TITLE
Feat clippy

### PR DESCRIPTION
**Pull request by zgfzgf**
fix clippy warning

### CODE REVIEW

**Comment by zgfzgf** [comment]
cargo 1.49.0 (d00d64df9 2020-12-05)
rustup 1.23.1 (3df2264a9 2020-11-30)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.49.0 (e1884a8e3 2020-12-29)`

cargo build --verbose --release --tests    OK

File: src/groth16/prover.rs [hunk 1]
```diff
@@ -313,18 +313,18 @@ where
...
     let mut g_c;
     {
         let mut rs = r;
         rs.mul_assign(&s);
 
-        g_c = vk.delta_g1 * &rs;
-        AddAssign::<&E::G1>::add_assign(&mut g_c, &(vk.alpha_g1 * &s));
-        AddAssign::<&E::G1>::add_assign(&mut g_c, &(vk.beta_g1 * &r));
+        g_c = vk.delta_g1 * rs;
+        AddAssign::<&E::G1>::add_assign(&mut g_c, &(vk.alpha_g1 * s));
```
Comment (str4d) [inline src/groth16/prover.rs:326]: These changes break compilation, so we can't address the clippy lint in this way.

File: src/groth16/generator.rs [hunk 2]
```diff
@@ -467,12 +466,12 @@ where
...
-        beta_g2: (g2 * &beta).to_affine(),
-        gamma_g2: (g2 * &gamma).to_affine(),
-        delta_g1: (g1 * &delta).to_affine(),
-        delta_g2: (g2 * &delta).to_affine(),
+        alpha_g1: (g1 * alpha).to_affine(),
+        beta_g1: (g1 * beta).to_affine(),
+        beta_g2: (g2 * beta).to_affine(),
+        gamma_g2: (g2 * gamma).to_affine(),
+        delta_g1: (g1 * delta).to_affine(),
+        delta_g2: (g2 * delta).to_affine(),
```
Comment (str4d) [inline src/groth16/generator.rs:474]: Ditto.

File: src/lib.rs [hunk 3]
```diff
@@ -135,7 +135,7 @@
 //! be separate crates that pull in the dependencies they require.
 
 // Catch documentation errors caused by code changes.
-#![deny(intra_doc_link_resolution_failure)]
+#![deny(broken_intra_doc_links)]
```
Comment (str4d) [inline src/lib.rs:138]: We cannot make this change until MSRV is at least 1.48.0 (see https://github.com/zcash/librustzcash/issues/279).

**Review by str4d — CHANGES_REQUESTED** [review]
Most of the changes LGTM and address lints we are now seeing after #64.

**Pull request closed without merging**